lib/Dialect/HW/HWOps.cpp

```diff
@@ -1243,21 +1243,24 @@ SmallVector<Attribute> HWModuleOp::getAllPortAttrs() {
   auto attrs = getPerPortAttrs();
   if (attrs && !attrs->empty())
     return {attrs->getValue().begin(), attrs->getValue().end()};
-  return SmallVector<Attribute>(getNumPorts());
+  return SmallVector<Attribute>(getNumPorts(),
+                                DictionaryAttr::get(getContext()));
```

Comment:
I wasn't supper happy with the idea of allocating new DictionaryAttrs (uniqing code path is a major runtime cost right now) just to return empty arrays.